### ISSUE TITLE
AirPlay: MediaControlPlayState never Finished

### ISSUE THREAD

**Issue by DevelopIdeas** [ConnectSDK/Connect-SDK-iOS#146]
When playing videos using AirPlay I'm trying to know when to automatically load the next video in my playlist. The problem is instead of MediaControlPlayStateFinished being returned at the end of playback MediaControlPlayPaused is returned instead.

Any suggestions?

**Comment by eunikolsky** [comment]
@DevelopIdeas, thanks for the report. Indeed, the parsing logic seems to be incorrect (https://github.com/ConnectSDK/Connect-SDK-iOS-Core/blob/dev/Services/Helpers/AirPlayServiceHTTP.m#L535-544).

## responses

Sample `playback-info` response when playing:
```js
{
    duration = "5.568";
    loadedTimeRanges =     (
                {
            duration = "5.568";
            start = 0;
        }
    );
    playbackBufferEmpty = 1;
    playbackBufferFull = 0;
    playbackLikelyToKeepUp = 1;
    position = "1.207569559";
    rate = 1;
    readyToPlay = 1;
    readyToPlayMs = 432;
    seekableTimeRanges =     (
                {
            duration = "5.568";
            start = 0;
        }
    );
    stallCount = 0;
    uuid = "C9E28C31-59FB-4F78-97C7-C8CF8960F243-1410-000003E741054B97";
}
```

Sample response when paused:
```js
{
    duration = "5.568";
    loadedTimeRanges =     (
                {
            duration = "5.568";
            start = 0;
        }
    );
    playbackBufferEmpty = 1;
    playbackBufferFull = 0;
    playbackLikelyToKeepUp = 1;
    position = "2.480788888888889";
    rate = 0;
    readyToPlay = 1;
    seekableTimeRanges =     (
                {
            duration = "5.568";
            start = 0;
        }
    );
    stallCount = 0;
    uuid = "C9E28C31-59FB-4F78-97C7-C8CF8960F243-1410-000003E741054B97";
}
```

The diff is (ignoring `position`):
```diff
13c13
<     rate = 1;
---
>     rate = 0;
15d14
<     readyToPlayMs = 432;
```

And sample response when the video is finished:
```js
{
    readyToPlay = 0;
    uuid = "C9E28C31-59FB-4F78-97C7-C8CF8960F243-1410-000003E741054B97";
}
```

When fast forwarding the rate is 2, and when rewinding it's -2.

## suggested fix

There is no explicit `state` property, so we should infer it somehow (a better way would be to use events, https://nto.github.io/AirPlay.html#video-events, https://github.com/ConnectSDK/Connect-SDK-iOS/issues/64). This should work:
* finished iff `rate` property is missing (and `readyToPlay == 0` ?);
* paused iff `rate == 0`;
* playing iff `rate != 0`;
* unknown otherwise.

Tests are required to cover the change.

**Comment by DevelopIdeas** [comment]
@eunikolsky thanks for the rapid response.

I've added your recommended fix and after some initial debugging everything looks good, I'll report back if I have any problems with the fix.

Thanks

**Comment by eunikolsky** [comment]
@DevelopIdeas, great that it works. I'm preparing a PR.

**Comment by eunikolsky** [comment]
Fixed in https://github.com/ConnectSDK/Connect-SDK-iOS/commit/1219d34146f07eb2f7b10ae1bc0feb76962fdc86 and https://github.com/ConnectSDK/Connect-SDK-iOS-Lite/commit/c687a60424e1a0cce064174b181d7ddfec5893d0

**Comment by DevelopIdeas** [comment]
@eunikolsky after further tests I've found that this fix does not work correctly, when starting to play a video, even before the video has actually started to play, MediaControlPlayStateFinished fires.

**Comment by eunikolsky** [comment]
@DevelopIdeas, indeed, it happens on long videos if you ask for play status while the video is being cached. The response is
```js
{
    readyToPlay = 0;
    uuid = "66FEA96A-D779-4AB9-A7CE-3E61A4C0B95C-1410-000005B05E4EA278";
}
```
which is the same as when the video is over.

**Comment by DevelopIdeas** [comment]
The same issue occurs when using a Google ChromeCast during a seek, the finished status gets incorrectly returned.
